From 5001e0b41a9ee22633b0a56571b2dc61bc55df4d Mon Sep 17 00:00:00 2001
From: Daniel Drake <dsd@laptop.org>
Date: Thu, 17 May 2012 20:21:00 +0000
Subject: Update FR OLPC layout

Some keys have only two characters printed: the unmodified character,
and a character in the Shift+AltGr position (upper right).
For such keys, users were confused why the "other" character could not
be accessed with AltGr alone.

Make such keys behave shift-invariant to remove this confusion.

Also add some obvious omissions from the previous submission fixing
various other keys that weren't working as printed.

The corresponding keyboard can be seen here:
http://wiki.laptop.org/go/OLPC_Azerty_Keyboard

Investigation and original patch from Walter Bender.

https://bugs.freedesktop.org/show_bug.cgi?id=50064
---
diff --git a/symbols/fr b/symbols/fr
index 85bcd55..45ef6a4 100644
--- a/symbols/fr
+++ b/symbols/fr
@@ -48,13 +48,17 @@ xkb_symbols "olpc" {
 
     name[Group1]="French";
 
-    key <AD02>	{ [ z,  Z,  VoidSymbol, guillemotleft ]	};
-
-    key <AC02>	{ [ s,  S,  VoidSymbol, ssharp ] };
-
-    key <AB02>	{ [  x,  X,  VoidSymbol,     guillemotright ] };
-
-    key <I219>  { [ less, greater ] };
+    key <I219>	{ [ less, greater ]	};
+    key <AD11>	{ [ dead_circumflex, dead_diaeresis, notsign, dead_abovering ]	};
+    key <AB08>	{ [ semicolon, period, underscore, multiply ]	};
+    key <TLDE>	{ [ twosuperior, asciitilde, VoidSymbol, VoidSymbol ]	};
+
+    // Some keys only have the Shift+AltGr character printed on them (alongside
+    // the unmodified one). Make such keys shift-invariant so that the printed
+    // value is achieved by pressing AltGr or Shift+AltGr.
+    key <AB02>	{ [ x,  X,  guillemotright, guillemotright ]	};
+    key <AC02>	{ [ s,  S,  ssharp, ssharp ]	};
+    key <AD02>	{ [ z,  Z,  guillemotleft, guillemotleft ]	};
 };
 
 partial alphanumeric_keys
--
cgit v0.9.0.2-2-gbebe
